depmapomics/ccle.py

```diff
@@ -0,0 +1,1423 @@
+    """
+    from taigapy import TaigaClient
+
+    tc = TaigaClient()
+
+    if prevcounts == "ccle":
+        prevcounts = tc.get(name=TAIGA_ETERNAL, file="CCLE_RNAseq_reads")
+
+    ccle_refsamples = trackerobj.read_tracker()
+    if gumbo:
+        ccle_refsamples = trackerobj.read_seq_table()
+
+    todrop += ccle_refsamples[
+        (ccle_refsamples.blacklist == 1) & (ccle_refsamples.datatype == "rna")
+    ].index.tolist()
+    priority = ccle_refsamples[
+        (ccle_refsamples.prioritized == 1) & (ccle_refsamples.datatype == "rna")
+    ].index.tolist()
+
+    folder = os.path.join("output", samplesetname, "")
+
+    if dry_run:
+        folder = os.path.join("output", "dryrun", "")
+
+    h.createFoldersFor(folder)
+    files, failed, _, renaming, lowqual, _ = await expressions.postProcess(
+        refworkspace,
+        samplesetname,
+        save_output=folder,
+        doCleanup=doCleanup,
+        priority=priority,
+        colstoclean=colstoclean,
+        ensemblserver=ensemblserver,
+        todrop=todrop,
+        samplesetToLoad=samplesetToLoad,
+        geneLevelCols=RSEMFILENAME_GENE,
+        trancriptLevelCols=RSEMFILENAME_TRANSCRIPTS,  # compute_enrichment=False,
+        ssGSEAcol="genes_tpm",
+        dropNonMatching=dropNonMatching,
+        dry_run=dry_run,
+        samplesinset=samplesinset,
+        rsemfilelocs=rsemfilelocs,
+        rnaqclocs=rnaqclocs,
+        **kwargs,
+    )
+
+    print("doing validation")
+    nonoverlap = set(prevcounts.columns) ^ set(
+        files.get("genes_expected_count").columns
+    )
+    print("number of non overlaping genes:")
+    print(len(nonoverlap))
+    # have we lost any samples compared to last release?
+    lost = set(prevcounts.index) - set(files.get("genes_expected_count").index)
+    print("of which, lost genes:")
+    print(lost)
+    # do we have samples that are missanotated compared to previous releases (replicate level)
+    # notindataset, missannotated, unmatched = findMissAnnotatedReplicates(replevel, prevcounts, renaming)
+    # for k,v in unmatched.items():
+    #    if ccle_refsamples.loc[k].arxspan_id!=v:
+    #        print(k,v)
+    # do we have samples that are missanotated compared to previous releases (sample level)
+    unmatched = rna.getDifferencesFromCorrelations(
+        files.get("genes_expected_count"), prevcounts, minsimi=minsimi
+    )
+    print("differences in correlations against the previous release")
+    print(unmatched)
+    # Is it because of  duplicate version?
+    print("do we see it as a duplicate in the tracker?")
+    rnasamples = ccle_refsamples[ccle_refsamples.datatype == "rna"]
+    for i, _ in unmatched:
+        print(len(rnasamples[rnasamples.arxspan_id == i]))
+
+    # CCLE_expression, CCLE_expression_full, ,
+    print("comparing to previous release")
+    # h.compareDfs(files["rsem_transcripts_tpm"], tc.get(name=TAIGA_ETERNAL, file='CCLE_RNAseq_transcripts'))
```

Comment:
same.

Reply:
I realized that the issue with the new indexing system is that the matrices pulled from eternal are all indexed by ModelID, whereas at this step of postprocessing all matrices are indexed by CDS-id. We can either create a parallel eternal dataset on taiga that hosts CDS-id (or profileID) indexed files, or move this comparison elsewhere. What do you think?

Reply:
let's move it elsewhere yes, in the virtual dataset creation maybe